src/Microsoft.DotnetOrg.Policies/Rules/PR18_ReposShouldNotUseDeprecatedBranchNames.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace Microsoft.DotnetOrg.Policies.Rules
+{
+    internal sealed class PR18_ReposShouldNotUseDeprecatedBranchNames : PolicyRule
+    {
+        public override PolicyDescriptor Descriptor { get; } = new PolicyDescriptor(
+            "PR18",
+            "Repos shouldn't use deprecated branch names",
+            PolicySeverity.Hidden
+        );
+
+        public override void GetViolations(PolicyAnalysisContext context)
+        {
+            var deprecatedBranchNames = new (string Deprecated, string Preferred)[]
+            {
+                ("master", "main")
+            };
+
+            foreach (var repo in context.Org.Repos)
+            {
+                if (repo.IsArchived || repo.IsSoftArchived())
+                    continue;
+
+                if (repo.IsFork)
+                    continue;
+
+                if (!repo.IsOwnedByMicrosoft())
+                    continue;
+
+                foreach (var branch in repo.Branches)
+                {
+                    foreach (var (deprecatedName, preferredName) in deprecatedBranchNames)
+                    {
+                        if (branch.Contains(deprecatedName, StringComparison.OrdinalIgnoreCase))
```

Comment:
I think a simple `Contains` is possibly going to run into problems. Consider someone who gets a bit proud and names their branch `masterwork`. Instead I would recommend we split on the common separators like `-` and `/` and then do an equality check.

Reply:
That's fair. When I ran the report it mostly found branches like these:

bevillal/master-before-16.10-branding
darc-master-feff497c-c731-4072-ac27-706a2d083a3e
dotnet-pr-tool-test/master
feed-update-master
github-dotnet/master
javiercn/merge-blazor-wasm-master
local-xa-master
master-backup
master-branch
master-links
master-pre-arcade
master-sxs
master-vs-deps
master-workaround-publishing-issue
MasterBeforeArcade
merges/dev16.0.x-to-master
merges/master-to-feature/witness-passing
merges/master-to-features/module-initializers
merges/master-to-post-dev15.5-contrib-20171030-065938
merges/master-to-release/6.0.1xx-preview1
merges/master-to-release/6.0.1xx-preview2
merges/master-to-release/6.0.1xx-preview3
merges/master-to-release/dev16.8
merges/release/dev16.7-vs-deps-to-master-vs-deps
merges/release/dev16.9-vs-deps-to-master-vs-deps
vramak/RemoveMaster

Some of them probably should be deleted, some of them probably should be renamed. Given that the policy tooling has an override mechanism I'm not too concerned with false positives, so long the rate is very low.

Thoughts?

Reply:
LGTM, I think we shouldn't be overzealous as this initiative should be primarily focused on changing the default branch name. There might be cases where the "master" string would be legitimate part of the branch name and we shouldn't make it a forbidden string.

Reply:
> There might be cases where the string "master" would be legitimate part of the branch name and we shouldn't make it a forbidden string.

The tool has a way to ignore false positives. When I ran the tool against the entire org, I didn't find any false positives though. For now my proposal would be: let's do a contains check, suppress false positives in case anything ever shows up and change the rule if it becomes unberable.